storagemarket/impl/provider.go

```diff
@@ -265,14 +265,9 @@ func (p *Provider) receiveDeal(s network.StorageDealStream) error {
 		return xerrors.Errorf("failed to get deal proposal from proposal message")
 	}
 
-	proposalNd, err := cborutil.AsIpld(proposal.DealProposal)
-	if err != nil {
-		return err
-	}
-
```

Comment:
I'm not seeing why this change was necessary. Unless it's just cleanup.

Reply:
The provider deal datastore is indexed by signed deal proposal cid.
If the Label field contains bytes that are not a utf8-compliant string, the CBOR-marshaled deal proposal will be different between the old format (label as a string) and the new format (DealLabel)
Therefore the signed deal proposal cid will also be different.

Lotus client
- generates the signed deal proposal cid
- sends a deal proposal
- reads the response
- checks that the signed deal proposal cid returned by the provider matches the one generated on the client

When the provider receives a deal proposal we need to know what format the deal proposal was in when the client CBOR-marshaled it, in order to be able to create the right signed deal proposal cid. `ReadDealProposal` checks the deal proposal protocol version to determine how to interpret the CBOR-marshaled deal proposal (old format vs new format)

Reply:
I backed out this change - instead we reject deal proposals over the v1.1.0 protocol where the deal label is not parseable utf8